gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/test/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAEnrollEndpointTest.java

```diff
@@ -0,0 +1,349 @@
+        smsFactor.setFactorType(FactorType.SMS);
+
+        when(factorManager.getFactor(ENROLL_FACTOR_ID)).thenReturn(emailFactor);
+        when(factorManager.get(ENROLL_FACTOR_ID)).thenReturn(mock(FactorProvider.class));
+
+        when(factorManager.getFactor(USER_FACTOR_ID)).thenReturn(smsFactor);
+        when(factorManager.get(USER_FACTOR_ID)).thenReturn(mock(FactorProvider.class));
+
+        router.route(HttpMethod.POST, "/mfa/enroll")
+                .handler(ctx -> {
+                    User user = new User();
+                    user.setId("userId");
+                    EnrolledFactor enrolledFactor = new EnrolledFactor();
+                    EnrolledFactorSecurity enrolledFactorSecurity = new EnrolledFactorSecurity();
+                    enrolledFactor.setFactorId(USER_FACTOR_ID);
+                    enrolledFactor.setStatus(FactorStatus.ACTIVATED);
+                    enrolledFactor.setSecurity(enrolledFactorSecurity);
+                    user.setFactors(Collections.singletonList(enrolledFactor));
+
+                    Client client = new Client();
+                    client.setFactors(Set.of(ENROLL_FACTOR_ID, USER_FACTOR_ID));
+                    ctx.setUser(io.vertx.reactivex.ext.auth.User.newInstance(new io.gravitee.am.gateway.handler.common.vertx.web.auth.user.User(user)));
+                    ctx.put(ConstantKeys.CLIENT_CONTEXT_KEY, client);
+
+                    ctx.next();
+                })
+                .handler(new MFAEnrollEndpoint(factorManager, engine, userService, domain))
+                .failureHandler(new ErrorHandler(RootProvider.PATH_ERROR));
+
+        testRequest(
+                HttpMethod.POST,
+                "/mfa/enroll",
+                req -> {
+                    Buffer buffer = Buffer.buffer();
+                    buffer
+                            .appendString("factorId="+ENROLL_FACTOR_ID)
+                            .appendString("&")
+                            .appendString(ConstantKeys.USER_MFA_ENROLLMENT+"="+true)
+                            .appendString("&")
+                            .appendString(ConstantKeys.MFA_ENROLLMENT_SHARED_SECRET+"="+UUID.randomUUID())
+                            .appendString("&")
+                            .appendString(ConstantKeys.MFA_ENROLLMENT_EMAIL+"="+EMAIL_ADDR);
+                    req.headers().set("content-length", String.valueOf(buffer.length()));
+                    req.headers().set("content-type", "application/x-www-form-urlencoded");
+                    req.write(buffer);
+                },
+                resp -> {
+                    String location = resp.headers().get("location");
+                    assertNotNull(location);
+                    assertTrue(location.contains("/error"));
+                    assertTrue(location.contains("factor+already+enrolled"));
+                },
+                HttpStatusCode.FOUND_302, "Found", null);
+    }
+
+  private static User createUser() {
```

Comment:
createUser() and createUserFactor() are unused

Reply:
Don't know why it is not used in this version, My branch is up to date and these methods are both used... :thinking:

Reply:
![image](https://github.com/gravitee-io/gravitee-access-management/assets/152597515/2ff0bfa4-5543-4d97-a7ff-14fcf1fbd773)
createUser() is unused and createUserFactor() is used only in createUser()

Reply:
indeed, was looking to the MFAChallengeEndpointTest class :face_exhaling: